src/NServiceBus.TransactionalSession.Tests/TransactionalSessionTests.cs

```diff
@@ -0,0 +1,141 @@
+﻿namespace NServiceBus.TransactionalSession.Tests;
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Extensibility;
+using Fakes;
+using NUnit.Framework;
+
+[TestFixture]
+public class TransactionalSessionTests
+{
+    [Test]
+    public async Task Open_should_generate_new_session_id()
```

Comment:
Could we share this and some other tests between `TransactionSession` and `OutboxTransactionalSession`?

Reply:
hm the only approach that comes to my mind immediately would be to create a custom subclass of `TransactionalSessionBase` that does nothing and then we could basically test the base class behavior but then we'd miss breaking behavior in the actual subclasses.

Or making some `TestCase` based tests that are testing the interface behavior against all provided implementations but that also sounds a bit excessive.

duplicating this is not ideal but I think it's good enough for now?